strawberry/utils/docstrings.py

```diff
@@ -0,0 +1,93 @@
+import inspect
+from typing import Any, Dict, Optional
+
+import docstring_parser
+from attributes_doc import get_attributes_doc
+
+
+class Docstring:
+    def __init__(self, target: Any) -> None:
+        self.target = target
+        self._docstring: Optional[docstring_parser.Docstring] = None
+        self._attribute_raw_docstrings: Optional[Dict[str, str]] = None
+        self._attribute_docstrings: Dict[str, Optional[str]] = {}
+
+    @property
+    def parsed_docstring(self) -> Optional[docstring_parser.Docstring]:
+        # Parse docstrings lazily, to avoid extra processing if
+        # config descriptions_from_docstrings is disabled
+        if self._docstring is None:
+            if isinstance(self.target, str):
+                self._docstring = docstring_parser.parse(self.target)
+            else:
+                self._docstring = docstring_parser.parse(self.target.__doc__ or "")
+
+            # if target is a class, fetch param docstrings from supertypes
+            if inspect.isclass(self.target):
+                for parent in self.target.mro():
+                    if parent is self.target or parent is object:
+                        continue
+                    if parent.__doc__ is None:
+                        continue
+                    parent_docstring = docstring_parser.parse(parent.__doc__)
+                    self._docstring.meta += parent_docstring.params
+
+        return self._docstring
+
+    @property
+    def main_description(self) -> Optional[str]:
+        docstring = self.parsed_docstring
+        if docstring is None:
+            return None
+
+        parts = []
+        if docstring.short_description:
+            parts.append(docstring.short_description)
+        if docstring.blank_after_short_description:
+            parts.append("")
+        if docstring.long_description:
+            parts.append(docstring.long_description)
+
+        # TODO: Expose other docstring bits (returns, raises, examples, etc)
+        return "\n".join(parts).strip() or None
```

Comment:
Yes, docstring-parser is not clever enough to remove the empty lines between the main description and the argument section, so we would end with a lot of unnecessary blank lines otherwise

Reply:
`long_description` is sometimes an empty string, so it would end with a trailing \n